lib/src/providers/VernaProvider/index.tsx

```diff
@@ -131,32 +157,32 @@ function VernaProvider({
   defaultLocale,
   defaultSchema,
   defaultSelector,
-  defaultUiSchema,
   defaultWidgets,
   intl,
   isEditor,
   locale,
   SubmitButton,
-  DropZone,
+  DropZone: DropZoneRender,
   FieldTemplate,
   Section,
   transformErrors,
-  translations,
 }: PropsWithChildren<VernaProviderProps>) {
-  // Both fullSchema & fullUiSchema are not used by the lib itself but may be
-  // used for further implementation
-  const [fullSchema, setFullSchema] = useState<VernaJSONSchemaType>(defaultSchema);
-  const [fullUiSchema, setFullUiSchema] = useState<UiSchema>(
-    cleanUiSchema(defaultSchema, defaultUiSchema, Section),
+  const [schema, _schemaDispatch] = useReducer<Reducer<VernaSchemaType, ReducerAction>>(
+    schemaReducer,
+    defaultSchema,
   );
+  const [selector, setSelector] = useState<string | undefined>(defaultSelector);
 
-  const [schema, _setSchema] = useState<VernaJSONSchemaType>(
-    getSelectedSchema(defaultSchema, defaultSelector),
-  );
-  const [uiSchema, _setUiSchema] = useState<UiSchema>(
-    getSelectedUiSchema(defaultUiSchema, defaultSelector),
-  );
-  const [schemaTranslations, setSchemaTranslations] = useState<TranslationType>(translations);
+  /**
+   * Overload of the schema dispatch to add the selector automatically
+   * @param action given to the dispatch
+   */
+  function schemaDispatch(action: ReducerAction) {
+    _schemaDispatch({ ...action, meta: { ...action.meta, selector } });
+  }
+
+  // A ajouter sur l'init de l'uischema cleanUiSchema(defaultSchema, defaultUiSchema, Section),
+  // \-> Ajouter en fonction d'init du reducer ?
```

Comment:
I did think about doing it (i even started it), but i wanted to talk about it with you before, except if i forgot something about it, it may be a good idea to clear things a bit ! :+1: